packages/beacon-node/src/chain/blocks/importBlock.ts

```diff
@@ -305,8 +305,18 @@ export async function importBlock(
   // MUST happen before any other block is processed
   // This adds the state necessary to process the next block
   this.stateCache.add(postState);
+
   await this.db.block.add(block);
 
+  if (this.config.getForkSeq(block.message.slot) >= ForkSeq.eip4844) {
+    if (!blobs) {
```

Comment:
should this extraction and error throw be before forkChoice.onBlock...

Reply:
yes, our import block flow is unsafe because it can end up in an inconsistent state